### PR TITLE
Added examples for `sentry:project:create` action and unit tests.

### PR DESCRIPTION
**Pull request by mondragonda**
## Hey, I just made a Pull Request!

Added examples for sentry project creation scaffolder action. And added the tests for the action code.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by acierto — APPROVED** [review]

File: plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts
Comment (freben) [inline plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts:45]: Not fond of mocking globals like this if we don't have to. Two options:

- Use `msw` to capture all network traffic and mock responses to it. That's a very popular method; check the repo for a ton of usage examples
- Change `createSentryCreateProjectAction` to accept an optional `fetch?: FetchApi` parameter (like [here](https://github.com/backstage/backstage/blob/2d39c1dbdf065c588464c6c0c4fc44bb469aeca0/plugins/graphiql/src/lib/api/GraphQLEndpoints.ts#L55)), and pass in a fake one in each test below

Again, msw is popular and extremely nice because it also ensures that there's absolutely no possibility of accidentally leaking network calls from the tests.

**Review by freben — COMMENTED** [review]
Thanks! Just a test fix suggestion

File: plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts
Comment (mondragonda) [reply plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts:45]: > - Use msw to capture all network traffic and mock responses to it. That's a very popular method; check the repo for a ton of usage examples

Agree.  I was reading about fetch calls not being caught by msw and the action uses node global fetch for creating the requests to Sentry. I will try to approach with msw. 

> - Change createSentryCreateProjectAction to accept an optional fetch?: FetchApi parameter (like [here](https://github.com/backstage/backstage/blob/2d39c1dbdf065c588464c6c0c4fc44bb469aeca0/plugins/graphiql/src/lib/api/GraphQLEndpoints.ts#L55)), and pass in a fake one in each test below

Agree. Will do.

Comment (mondragonda) [reply plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts]: @freben Updated with the parameter added.

**Comment by freben** [comment]
Seems you'll want to rebase on a fresh master; some conflict crept in.

**Comment by mondragonda** [comment]
@freben I think is done now

**Comment by mondragonda** [comment]
@backstage-service does this PR should be merged?

**Comment by freben** [comment]
The maintainers are off this week - we'll get back on the reviews soon!

File: plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts
Comment (freben) [inline plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts:48]: Mocking fetch is a bit unfortunate for a couple of reasons, one of which being that the Response object is a bit of a pain to deal with and fake.

It's pretty established within the project to lean heavily on the `msw` library for network mocking. See for example [what the confluence module does](https://github.com/backstage/backstage/blob/1b7d382f8f4a79e357f769b676f9b172dd342e94/plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/confluenceToMarkdown.examples.test.ts#L136) in its tests. The benefit is that you detach yourself from worrying about how fetch works, and you have this entire nice platform to leverage for how to fake various network calls. As a bonus, when using the [`setupRequestMockHandlers`](https://github.com/backstage/backstage/blob/1b7d382f8f4a79e357f769b676f9b172dd342e94/plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/confluenceToMarkdown.examples.test.ts#L47) helper, you ensure that no traffic in the test can ever accidentally leave the test worker.

File: plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts
Comment (freben) [inline plugins/scaffolder-backend-module-sentry/src/actions/createProject.test.ts:85]: When switching to msw, you can put expects directly _inside the request handler_ and then putting an `expect.assertions(2)` or whatever in the test to assert that that assertion (1) really does get exercised as part of the test.

File: plugins/scaffolder-backend-module-sentry/src/actions/createProject.ts
Comment (freben) [inline plugins/scaffolder-backend-module-sentry/src/actions/createProject.ts:36]: as suggested elsewhere, you should probably remove this change to to the creation function itself for now

**Review by freben — COMMENTED** [review]
Thanks! Some small comments

**Comment by camilaibs** [comment]
> @freben I think is done now

Could you please sync with the master once more 🙏🏻 ?

**Comment by freben** [comment]
rudely rebased, let's see how it goes

**Review by freben — APPROVED** [review]
Alright, updated the test to use msw too

**Comment by freben** [comment]
Man I'm mystified by these build errors. I don't get them locally. Gonna have to get back to this later it seems.

**Comment by freben** [comment]
alright one more try, rewriting the test to use msw2 😅

**Comment by mondragonda** [comment]
Sorry guys. I've been busy with work. I'll update when I have some time.

**Comment by freben** [comment]
No worries, this seems to have been some REALLY weird msw problem that was env specific, not your fault. Seems to be about to go green now, I'll merge later on if it does

**Pull request merged**
